src/jit/codegenxarch.cpp

```diff
@@ -718,7 +718,7 @@ void CodeGen::genCodeForDivMod(GenTreeOp* treeNode)
         // Note that a reg optional operand is a treated as a memory op
         // if no register is allocated to it.
         assert(divisor->isUsedFromReg() || divisor->isMemoryOp() || divisor->IsCnsFltOrDbl() ||
-               divisor->IsRegOptional());
+               divisor->IsRegOptional() || divisor->IsLocal());
```

Comment:
That is a very interesting question - I suspect that, at this point, this is really a useless assert. I'll look into it a bit.